### ISSUE TITLE
PHP 7.2 end of life

### ISSUE THREAD

**Issue by ybelenko** [OpenAPITools/openapi-generator#9200]
PHP 7.2 reached end of life at 30 Nov 2020

> End of life - A release that is no longer supported. Users of this release should upgrade as soon as possible, as they may be exposed to unpatched security vulnerabilities.

### [PHP: Supported Versions](https://www.php.net/supported-versions.php):
- ~7.2 until 30 Nov 2020~
- 7.3 until 6 Dec 2021
- 7.4 until 28 Nov 2022
- 8.0 until 26 Nov 2023

It seems that we need to change minimum PHP version to 7.3 in all php generators.

### Current Requirements:
| Generator | PHP version |
|:---------:|:------------:|
| ~php client~ | >= 7.3 \|\| >= 8.0         |
| Laravel server | >= 7.2.5      |
| Lumen server | >= 7.2.5        |
| ~Slim4 server~ | >= 7.3             |
| **Symfony server** | >= **7.1.3**     |
| ~Ze-ph server~ | >= 7.3            |

cc @jebentier @dkarlovi @mandrean @jfastnacht @ackintosh @renepardon

**Comment by ybelenko** [comment]
Before I start to update PHP version in generators I would like to know what community thinks about 8.0 support.
1. Can we say that code which works under 7.3 could also works under 8.0?
2. Is there any appropriate/recommended TravisCI config to test against 8.0? I just checked Travis docs, seems like there is no Ubuntu environment with preinstalled php 8.0. Neither mentions about 8.0 in general.
3. Is composer constraint `"php": "^7.3 || ^8.0"` correct?

cc @Articus as active PHP contributor

**Comment by Articus** [comment]
>     1. Can we say that code which works under 7.3 could also works under 8.0?

I do not think that it is safe to assume that - PHP 8.0 is a major release after all with [few backward incompatible changes](https://www.php.net/manual/en/migration80.incompatible.php) and even if generated code runs fine on 8.0 its dependencies may not support it yet.

>     2. Is there any appropriate/recommended TravisCI config to test against 8.0? I just checked Travis docs, seems like there is no Ubuntu environment with preinstalled php 8.0. Neither mentions about 8.0 in general.

According [official docs](https://docs.travis-ci.com/user/languages/php/#php-versions) they seem to allow PHP 8.0 for Ubuntu >= 16.04 as on-demand installation and it should be available with `language: php` environment. But I stopped using Travis CI about half a year ago because of forced migration from travis-ci.org so do not have any real experience with using PHP 8.0 there. 
 
>     3. Is composer constraint `"php": "^7.3 || ^8.0"` correct?

Yep, it is correct. There is another option `"php": >=7.3` (used by Symfony for example) but it requires a lot more proactive testing and support to check and maintain compatibility against beta releases of upcoming PHP versions.

**Comment by ybelenko** [comment]
Well, I gonna update `readme` and `composer.json` in all generators to `^7.4`. Also it's time to remove 7.3 environment from all PHP test configs(TravisCI), it should reduce CI execution time significantly.

Let's say that 8.0 version is beyond the project scope right now.

**Comment by ybelenko** [comment]
I know that our committee members are very busy, but guys please help me to choose composer constraint for all PHP generators from:
1. `"php": "^7.3"`. Add 🎉 reaction to vote for this option
2. `"php": "^7.3 || ^8.0"` with short notice in README that codegen hasn't been tested against 8.0 and should be considered unstable. Add 🚀 reaction to vote for this option.
3. Other option. Vote with 😕 reaction and leave your suggestion in comments.

**Comment by dominik-riha** [comment]
@ybelenko given that PHP 7.3 already now doesn't have active support and security fixes are being provided only until December **this year** - is there a reason why not directly bump the generated code to 7.4? We would benefit not only from typed function params (as of 7.3) but also from typed properties. E.g. looking at the [PetApi example](https://github.com/OpenAPITools/openapi-generator/blob/a0c3db1157fc756a5396c3c196badf24f3721fd8/samples/client/petstore/php/OpenAPIClient-php/lib/Api/PetApi.php#L54)
```
protected ClientInterface $client;
```
instead of
```
/**
 * @var ClientInterface
 */
protected $client;
```

**Comment by ybelenko** [comment]
@dominik-riha Agree. If I'm the only user of generator I would go with `^7.4` hands down. I'm just afraid that we can face frustration from current 7.3 php users, because we force them to upgrade PHP or edit `composer` template.

> ```
> protected ClientInterface $client;
> ```

Imho, class typing is unusable in PHP so far, I got major issues with `function(): self` constraints within classes with inheritance. So, I'm not sure that I can write stable codegen with class annotations yet 😄.

**Comment by chrillep** [comment]
> Laravel server	>= 7.2.5

@ybelenko as of Laravel 8.x
https://github.com/laravel/framework/blob/6216534f14c7cb8bdfa213e5f7e3d357119a4579/composer.json#L17-L18

in Laravel 9 it will be >= 8.x

**Comment by ybelenko** [comment]
Closed in favour of #11038
